b2/account_info.py

```diff
@@ -0,0 +1,318 @@
+        self.data[self.API_URL] = api_url
+        self.data[self.APPLICATION_KEY] = application_key
+        self.data[self.REALM] = realm
+        self.data[self.DOWNLOAD_URL] = download_url
+        self.data[self.MINIMUM_PART_SIZE] = minimum_part_size
+        self._write_file()
+
+    def set_bucket_upload_data(self, bucket_id, upload_url, upload_auth_token):
+        self.data[self.BUCKET_UPLOAD_DATA][bucket_id] = {
+            self.BUCKET_UPLOAD_URL: upload_url,
+            self.BUCKET_UPLOAD_AUTH_TOKEN: upload_auth_token,
+        }
+        self._write_file()
+
+    def get_bucket_upload_data(self, bucket_id):
+        bucket_upload_data = self.data[self.BUCKET_UPLOAD_DATA].get(bucket_id)
+        if bucket_upload_data is None:
+            return None, None
+        url = bucket_upload_data[self.BUCKET_UPLOAD_URL]
+        upload_auth_token = bucket_upload_data[self.BUCKET_UPLOAD_AUTH_TOKEN]
+        return url, upload_auth_token
+
+    def clear_bucket_upload_data(self, bucket_id):
+        self.data[self.BUCKET_UPLOAD_DATA].pop(bucket_id, None)
+
+    def set_large_file_upload_data(self, file_id, upload_url, upload_auth_token):
+        self.large_file_uploads[file_id] = (upload_url, upload_auth_token)
+
+    def get_large_file_upload_data(self, file_id):
+        return self.large_file_uploads.get(file_id, (None, None))
+
+    def clear_large_file_upload_data(self, file_id):
+        if file_id in self.large_file_uploads:
+            del self.large_file_uploads[file_id]
+
+    def save_bucket(self, bucket):
+        names_to_ids = self.data[self.BUCKET_NAMES_TO_IDS]
+        if names_to_ids.get(bucket.name) != bucket.id_:
+            names_to_ids[bucket.name] = bucket.id_
+            self._write_file()
+
+    def refresh_entire_bucket_name_cache(self, name_id_iterable):
+        names_to_ids = self.data[self.BUCKET_NAMES_TO_IDS]
+        new_cache = dict(name_id_iterable)
+        if names_to_ids != new_cache:
+            self.data[self.BUCKET_NAMES_TO_IDS] = new_cache
+            self._write_file()
+
+    def remove_bucket_name(self, bucket_name):
+        names_to_ids = self.data[self.BUCKET_NAMES_TO_IDS]
+        if bucket_name in names_to_ids:
+            del names_to_ids[bucket_name]
+        self._write_file()
+
+    def get_bucket_id_or_none_from_bucket_name(self, bucket_name):
+        names_to_ids = self.data[self.BUCKET_NAMES_TO_IDS]
+        return names_to_ids.get(bucket_name)
+
+    def _write_file(self):
+        flags = os.O_WRONLY | os.O_CREAT | os.O_TRUNC
+        if os.name == 'nt':
+            flags |= os.O_BINARY
+        with os.fdopen(os.open(self.filename, flags, stat.S_IRUSR | stat.S_IWUSR), 'wb') as f:
+            # is there a cleaner way to do this that works in both Python 2 and 3?
+            json_bytes = json.dumps(self.data, indent=4, sort_keys=True).encode('utf-8')
+            f.write(json_bytes)
+
+
+class StubAccountInfo(AbstractAccountInfo):
```

Comment:
I think this should be kept somewhere in `tests/` - we don't want to expose it to the user next to `StoredAccountInfo`